extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesAndWorkerTaskRunnerFactory.java

```diff
@@ -43,7 +46,7 @@ public class KubernetesAndWorkerTaskRunnerFactory implements TaskRunnerFactory<K
   public KubernetesAndWorkerTaskRunnerFactory(
       KubernetesTaskRunnerFactory kubernetesTaskRunnerFactory,
       HttpRemoteTaskRunnerFactory httpRemoteTaskRunnerFactory,
-      RemoteTaskRunnerFactory remoteTaskRunnerFactory,
+      @Nullable RemoteTaskRunnerFactory remoteTaskRunnerFactory,
```

Comment:
The pattern I've usually seen here is to inject a `Provider<X>` into the constructor and then use the config to decide whether or not to call `Provider.get()` https://github.com/google/guice/wiki/InjectingProviders#injecting-providers. 

Would this technique work? It seems slightly easier to follow as `RemoteTaskRunnerFactory` is already injectable from another module, so I'm not certain what the expected behavior is when 2 modules try to inject the same class.

Reply:
Agree, the provider pattern is simpler. Updated the code.